contrib/gogearbox/gearbox.v1/internal/gearboxutil/fashttpcarrier.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package gearboxutil
```

Comment:
We should remove the commented code before merging the PR. Taking a look at the existing HTTP implementation (https://github.com/DataDog/dd-trace-go/blob/main/ddtrace/tracer/textmap.go#L30-L44) - we could just make the same decissions:
- use `Set` to override headers instead of `Add`
- use the `ForeachKey` implementation you provided as it should have similar performance as the one from HTTP.

Reply:
We had the same discussion on an old commit.. I didn't resolve the convo but it's no longer showing up here? 
See: https://github.com/DataDog/dd-trace-go/pull/2118#discussion_r1296353750
You mentioned overwriting the key, and I basically said:

- FasthttpCarrier is not a HTTPHeadersCarrier type
- FasthttpCarrier fulfills TextMapCarrier (whose definition of Set is nested and very vague)
- grpc's [mdcarrier]( https://github.com/DataDog/dd-trace-go/blob/main/contrib/google.golang.org/internal/grpcutil/mdcarrier.go#L32-L37), which fulfills TextMapCarrier, appends rather than overwrites: https://github.com/DataDog/dd-trace-go/blob/main/contrib/google.golang.org/internal/grpcutil/mdcarrier.go#L32-L37

In short, the question is still unanswered IMO: **Do we want to append or set/overwrite data?**